src/options.rs

```diff
@@ -0,0 +1,215 @@
+use clap::ArgMatches;
+// use ignore::WalkBuilder;
+// use globset::{Glob, GlobSet, GlobSetBuilder};
+//use std::path::PathBuf;
+use std::ffi::OsString;
+
+#[derive(Debug)]
+pub struct CLIOptions {
+    pub included_path_matches: Vec<OsString>,
+    pub excluded_path_matches: Vec<OsString>,
+    pub destination_folder: OsString,
+    pub list_only_short: bool,
+    pub output_to_pipe: bool,
+    pub test_files_only: bool,
+    pub display_filename_only: bool,
+    pub no_overwrite: bool,
+    pub overwrite_without_asking: bool,
+    pub quiet: bool,
+    pub very_quiet: bool,
+}
+
+impl CLIOptions {
+    pub fn from_clap_matches(matches: &ArgMatches) -> CLIOptions {
+        let included_path_matches = get_paths_from_clap(matches, "files");
+
+        if included_path_matches.is_empty() {
+            panic!("You need to provide a valid path to a compressed file.");
+        }
+
+        let excluded_path_matches = get_paths_from_clap(matches, "exclude_files");
+
+        let destination_folder: OsString = matches
+            .value_of_os("extract_directory")
+            .map(OsString::from)
+            .unwrap_or_else(|| OsString::from("./"));
+
+        CLIOptions {
+            included_path_matches,
+            excluded_path_matches,
+            destination_folder,
+            list_only_short: matches.is_present("list_archives"),
+            output_to_pipe: matches.is_present("extrat_to_pipe"),
+            test_files_only: matches.is_present("test_files"),
+            display_filename_only: matches.is_present("display_only"),
+            no_overwrite: matches.is_present("no_overwrite"),
+            overwrite_without_asking: matches.is_present("overwrite_no_prompt"),
+            quiet: matches.is_present("quiet"),
+            very_quiet: matches.occurrences_of("quiet") == 2,
+        }
+    }
+
+    pub fn process_options(&self) -> RZipOptions {
+        let output = if self.output_to_pipe {
+            RZipOutput::StdOut
+        } else {
+            RZipOutput::File(self.destination_folder.clone())
+        };
+        let verbose = if self.quiet && self.very_quiet {
```

Comment:
you can do another match, it might be a biy more readable.

Reply:
When matching bools clippy tells me to not use match. I opted to just follow it. I see that for the ones with 3 branches I can do it with math. Changing code...

Reply:
Might be, not sure if a Tuple of boolean also matches that lint.
`match (self.quite, self.very_quite)` nice trick you can do with matches :grin: